### ISSUE TITLE
"default" field in PushRule omitted for user-defined push rules

### ISSUE THREAD

**Issue by Karlinde** [matrix-org/synapse#6636]
### Description

The [spec ](https://matrix.org/docs/spec/client_server/r0.5.0#get-matrix-client-r0-pushrules) states that for push rules returned from the server, the field `default` is required. Synapse however seems to omit this field for user-defined rules, leading to deserialization failures for a spec-compliant client that expects it when fetching push rules.

### Steps to reproduce

- Push a user-defined push rule to synapse using `PUT /_matrix/client/r0/pushrules/{scope}/{kind}/{ruleId}`.
- Fetch all push rules using `GET /_matrix/client/r0/pushrules/`.
- Find the newly created rule in the returned ruleset. It doesn't have a `default` field.

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from _homeserver.log_, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.
-->

Since the spec states that the `default` field is required, I did not expect the server to be allowed to omit it.

### Version information

<!-- IMPORTANT: please answer the following questions, to help us narrow down the problem -->

<!-- Was this issue identified on matrix.org or another homeserver? -->
- **Homeserver**: private, not federating

<!--
 What version of Synapse is running?

You can find the Synapse version with this command:

$ curl http://localhost:8008/_synapse/admin/v1/server_version

(You may need to replace `localhost:8008` if Synapse is not configured to
listen on that port.)
-->
- **Version**: 1.7.3

- **Install method**: Ansible playbook from https://github.com/spantaleev/matrix-docker-ansible-deploy
<!-- examples: package manager/git clone/pip  -->

- **Platform**: Docker in Ubuntu running on a VPS.
<!--
Tell us about the environment in which your homeserver is operating
distro, hardware, if it's running in a vm/container, etc.
-->

**Comment by Karlinde** [comment]
Looking at the code, it looks like the issue is that only the built-in rule objects ever get the "default" key set, which seems to happen [here](https://github.com/matrix-org/synapse/blob/63829145870473fecdca45ba2bdb9fa0a8eae06a/synapse/push/baserules.py#L361).

A simple solution would be to just add `rule["default"] = False` for each rule in [`_load_rules()`](https://github.com/matrix-org/synapse/blob/77661ce81a799a375317dff9e4c8696da528984c/synapse/storage/data_stores/main/push_rule.py#L43).

**Comment by jplatte** [comment]
Fixed in #6639?

**Comment by clokep** [comment]
Looks like we never closed this after merging the PR. I'm going to close this, but let us know if there's more to do here!
